lib/ancestry/class_methods.rb

```diff
@@ -17,6 +17,17 @@ def scope_depth depth_options, depth
       end
     end
 
+    # Scope that returns all the leaves
+    def leaves
+      id_column = "#{table_name}.id"
+      id_column_as_text = sql_cast_as_text(id_column)
+      parent_ancestry = sql_concat("#{table_name}.#{ancestry_column}", "'/'", id_column_as_text)
+
+      joins("LEFT JOIN #{table_name} AS c ON c.#{ancestry_column} = #{id_column_as_text} OR c.#{ancestry_column} = #{parent_ancestry}").
```

Comment:
is there a way to use an existing scope here?
Only adding the `group` and `having`

Also, should `id` be `primary_key`?

Reply:
better yet, is there a way to extract this concept so we can use it more?